pkg/azureutils/azure_disk_utils.go

```diff
@@ -254,36 +245,8 @@ func GetCloudProviderFromClient(ctx context.Context, kubeClient *clientset.Clien
 	return az, nil
 }
 
-// GetCloudProviderFromConfig get Azure Cloud Provider
-func GetCloudProvider(ctx context.Context, kubeConfig, secretName, secretNamespace, userAgent string,
-	allowEmptyCloudConfig, enableTrafficMgr bool, trafficMgrPort int64) (*azure.Cloud, error) {
-	kubeClient, err := GetKubeClient(kubeConfig)
-	if err != nil {
-		klog.Warningf("get kubeconfig(%s) failed with error: %v", kubeConfig, err)
-		if !os.IsNotExist(err) && !errors.Is(err, rest.ErrNotInCluster) {
-			return nil, fmt.Errorf("failed to get KubeClient: %v", err)
-		}
-	}
-	return GetCloudProviderFromClient(ctx, kubeClient, secretName, secretNamespace, userAgent,
-		allowEmptyCloudConfig, enableTrafficMgr, trafficMgrPort)
-}
-
-// GetKubeConfig gets config object from config file
-func GetKubeConfig(kubeconfig string) (config *rest.Config, err error) {
-	if kubeconfig != "" {
-		if config, err = clientcmd.BuildConfigFromFlags("", kubeconfig); err != nil {
-			return nil, err
-		}
-	} else {
-		if config, err = rest.InClusterConfig(); err != nil {
```

Comment:
where is this logic now in the new code ? 

```
if config, err = rest.InClusterConfig(); err != nil {
```

Reply:
clientcmd.BuildConfigFromFlags has similar logic https://github.com/kubernetes/client-go/blob/v0.29.0/tools/clientcmd/client_config.go#L616